tests/test_app_models.py

```diff
@@ -0,0 +1,163 @@
+
+    def test_token_generation(self):
+        """Test that token are generated in bytes."""
+        self.assertIsInstance(self.user.generate_auth_token(34), bytes)
+
+    def test_token_verification(self):
+        """Test verification of token.
+
+        Valid tokens return True while invalid tokens return False.
+        """
+        db.session.add(self.user, self.user2)
+        db.session.commit()
+        user_token = self.user.generate_auth_token(1)
+        self.assertNotEqual(self.user.verify_auth_token(user_token),
+                            self.user2)
+        self.assertEqual(self.user.verify_auth_token(user_token), self.user)
+        sleep(2)
+        self.assertIsNone(self.user.verify_auth_token(user_token))
+        self.assertIsNone(self.
+                          user.verify_auth_token('jdjdje230920093944334j'))
+
+    def test_shorturl_time_saving(self):
+        """Test that values saved to the when columns are datetime objects."""
+        self.long_url.short_urls.append(self.short_url)
+        self.user.short_urls.append(self.short_url)
+        self.long_url.users.append(self.user)
+        db.session.add(self.short_url)
+        db.session.commit()
+        self.assertIs(type(self.short_url.when), datetime)
+
+    def test_saving_user_to_db(self):
+        """Test that the save method of the user model saves to database."""
+        self.user.save()
+        db_user = User.query.filter_by(email='seni@andela.com').first()
+        self.assertIs(self.user, db_user)
+        self.assertIsInstance(db_user, User)
+
+    def test_shorturl_not_saving_without_longurl(self):
+        """Test that ShortUrl table does not accept Null values."""
+        db.session.add(self.short_url)
+        with self.assertRaises(sqlalchemy.exc.IntegrityError):
+            db.session.commit()
+
+    def test_shorturl_is_active_defualts(self):
+        """Test that ShortUrl is_active column is set to True by default."""
+        self.long_url.short_urls.append(self.short_url)
+        self.user.short_urls.append(self.short_url)
+        self.long_url.users.append(self.user)
+        db.session.add(self.short_url)
+        db.session.commit()
+        self.assertTrue(self.short_url.is_active)
+        self.short_url.is_active = False
+        self.assertFalse(self.short_url.is_active)
+
+    def test_shorturl_longurl_relationship(self):
+        """Test the many to one relationship between ShortUrl and LongUrl."""
+        self.long_url.short_urls.append(self.short_url)
+        self.long_url.short_urls.append(self.short_url2)
+        self.user.short_urls.append(self.short_url)
+        self.user.short_urls.append(self.short_url2)
+        self.long_url.users.append(self.user)
+        self.long_url2.short_urls.append(self.short_url)
+        self.long_url2.users.append(self.user)
+        db.session.add_all([self.user, self.short_url, self.short_url2,
+                            self.long_url, self.long_url2])
```

Comment:
So I have tried to ignore this but I see that it is cumbersome and possibly unnecessary. If a user has a long URL, this long URL could have several short URLs. A user is related to a long URL and a long URL is related to short URL. My thinking is if I have the appended a short URL to the long URL then I should have access to the short URL from the user model or am I wrong?